client/packages/system/src/Name/ListView/ImportProperties/UploadTab.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { FC, useState } from 'react';
+import Papa, { ParseResult } from 'papaparse';
+import { InlineProgress, Typography, Upload } from '@common/components';
+import { ImportPanel } from './ImportPanel';
+import { useTranslation } from '@common/intl';
+import {
+  Grid,
+  Stack,
+  Link,
+  FileUtils,
+  PropertyNode,
+  useNotification,
+  NamePropertyNode,
+} from '@openmsupply-client/common';
+
+import { ImportRow } from './PropertiesImportModal';
+import { importFacilitiesPropertiesToCsv } from '../utils';
+import { FacilityNameRowFragment } from '../../api/operations.generated';
+import { processProperties } from '../../../utils';
+
+interface UploadTabProps {
+  setFacilityProperties: React.Dispatch<React.SetStateAction<ImportRow[]>>;
+  setErrorMessage: (value: React.SetStateAction<string>) => void;
+  facilities: FacilityNameRowFragment[] | undefined;
+  onUploadComplete: () => void;
+  properties: NamePropertyNode[] | undefined;
+}
+interface ParsedImport {
+  id: string;
+  [key: string]: string | undefined;
+}
+
+enum FacilitiesColumn {
+  CODE = 0,
+  NAME = 1,
+}
+
+const getCell = (row: ParsedImport, index: FacilitiesColumn) => {
+  const rowKeys = Object.keys(row);
+  const key = rowKeys[index] ?? '';
+  return row[key] ?? '';
+};
+
+export const UploadTab: FC<ImportPanel & UploadTabProps> = ({
+  tab,
+  facilities,
+  setErrorMessage,
+  setFacilityProperties,
+  onUploadComplete,
+  properties,
+}) => {
+  const t = useTranslation();
+  const { error } = useNotification();
+  const [isLoading, setIsLoading] = useState(false);
+  const FacilityPropertyBuffer: ImportRow[] = [];
+  // TODO filter name properties for facility properties?
```

Comment:
It's fine for now, as we only use name properties for facility properties... I think either remove comment or mention that its a future improvement if we start using name properties in more places?